### PR TITLE
added replication parameters to puppet-redis configurator

### PR DESCRIPTION
**Pull request by wmstantoniii**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -127,22 +147,26 @@
...
+     redis_slowlog_max_len                                => $redis_slowlog_max_len,
+     redis_password                                       => $redis_password,
+     redis_is_slave                                       => $redis_is_slave,
+     redis_slaveof_master_ip                              => $redis_slaveof_master_ip,
+     redis_slaveof_master_port                            => $redis_slaveof_master_port,
+     redis_repl_backlog_size                              => $redis_repl_backlog_size,
+     redis_slave_priority                                 => $redis_slave_priority,
+     redis_slave_output_buffer_hard_limit                 => $redis_slave_output_buffer_hard_limit,
+     redis_slave_output_buffer_soft_limit                 => $redis_slave_output_buffer_soft_limit,
+     redis_slave_output_buffer_soft_limit_max_interval    => $redis_slave_output_buffer_soft_limit_max_interval,
```
Comment (SegFaultAX) [inline manifests/init.pp]: Our style dictates that there should be exactly 1 space after the longest key in the argument list, and every other arrow is aligned with that. It looks like you have about 3 extra spaces here.

Comment (wmstantoniii) [inline manifests/init.pp]: I deleted the offending spaces.  Now there should be exactly 1 space after the longest key in the argument list

File: manifests/init.pp [hunk 2]
```diff
@@ -127,22 +147,26 @@
 
   # Install default instance
   redis::instance { 'redis-default':
```
Comment (SegFaultAX) [inline manifests/init.pp]: Looks like the `redis::instance` class wasn't updated to accept these parameters, so you'll need to add them there as well.

## Test Case

With node definition:

```puppet
# Vagrant node def
node 'ubuntu2.vagrant.local' {
  class { '::redis':
    redis_repl_backlog_size => "backlog-size",
    redis_slave_output_buffer_hard_limit => "hard-limit",
    redis_slave_output_buffer_soft_limit => "soft-limit",
    redis_slave_output_buffer_soft_limit_max_interval => "interval"
  }
}
```

Error:

```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Invalid parameter redis_repl_backlog_size on Redis::Instance[redis-default] at /etc/puppet/modules/redis/manifests/init.pp:170 on node ubuntu2.vagrant.local
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```

**Comment by wmstantoniii** [comment]
I modified the redis::instance class to add the new parameters

**Comment by SegFaultAX** [comment]
LGTM.

:shipit: :hankey: :trollface:

**Pull request merged**